src/front/app/features/challenge/challenge.reducer.js

```diff
@@ -12,8 +14,12 @@ const reducer = (state = initialState, action = {}) => {
         return upsertChallenge(action.challenge, state);
     case RESET_STATE:
         return initialState;
-    case GET_USER:
+    case USER_RECEIVED:
         return upsertCurrentUser(action.user, state);
+    case ACCEPTED_RECEIVED:
```

Comment:
@borys-ihnatyev, ok I see your point. Sounds reasonable. I'll think about how to avoid redundant state changes.